packages/world/src/factories/WorldFactory.sol

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import { Create2 } from "./Create2.sol";
+import { World } from "../World.sol";
+import { IWorldFactory } from "./IWorldFactory.sol";
+import { IBaseWorld } from "../interfaces/IBaseWorld.sol";
+import { IModule } from "../interfaces/IModule.sol";
+
+contract WorldFactory is IWorldFactory {
+  IModule public coreModule;
+  uint256 public worldCount;
+
+  constructor(IModule _coreModule) {
+    coreModule = _coreModule;
+  }
+
+  /**
+    @dev Deploy a new World and install coreModule.
+  */
+  function deployWorld() public {
+    bytes memory bytecode = type(World).creationCode;
+    address worldAddress = Create2.create2Deploy(bytecode, worldCount);
+    IBaseWorld world = IBaseWorld(worldAddress);
+    world.installRootModule(coreModule, new bytes(0));
+    emit WorldDeployed(worldAddress);
+    worldCount++;
+  }
```

Comment:
Should we transfer the ownership of the root module to the caller after installing the root module? (Or maybe we could make it optional and emit the selected option as part of the event? The interesting thing with a deploy function that does not transfer ownership of the root module is that users can be sure the World behaves exactly as the audited vanilla World)

Reply:
As it currently stands it would be the Factory address that would be the owner? I imagine the caller would probably assume they will be the owner so maybe that is best although I'm happy to take direction as I don't have a full understanding of the knock on effects.